src/lib/y2network/interface_config_builder.rb

```diff
@@ -0,0 +1,160 @@
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+require "yast"
+require "y2network/interface_defaults"
+
+module Y2Network
+  # Stores what's needed when creating a new configuration for an interface
+  class InterfaceConfigBuilder
+    # @return [String] Device name (eth0, wlan0, etc.)
+    attr_accessor :name
+    # @return [String] type which is intended to be build
+    attr_accessor :type
+
+    include Y2Network::InterfaceDefaults
+
+    # Constructor
+    #
+    # Load with reasonable defaults;
+    # see LanItems::new_item_default_options, LanItems::@SysconfigDefaults and
+    # others as in LanItems::Select
+    def initialize
+      @config = init_device_config({})
+      @s390_config = init_device_s390_config({})
+    end
+
+    # Stores one option value tuple
+    # FIXME: obsolete by []=
+    def set(option: nil, value: nil)
+      @config[option] = value
+    end
+
+    # Returns currently stored option value
+    # FIXME: obsolete by []
+    def option(option)
+      @config.fetch(option, nil)
+    end
+
+    def []=(key, value)
+      @config[key] = value
+    end
+
+    def [](key)
+      @config[key]
+    end
+
+    # Provides stored configuration in sysconfig format
+    #
+    # @return [Hash<String, String>] where key is sysconfig option and value is the option's value
+    def device_sysconfig
+      # with naive implementation of filtering options by type
+      config = @config
+
+      # initalize options which has to be known and was not set by the user explicitly
+      init_mandatory_options
+
+      # filter out options which are not needed
+      config = config.delete_if { |k, _| k =~ /WIRELESS.*/ } if type != "wlan"
+      config = config.delete_if { |k, _| k =~ /BONDING.*/ } if type != "bond"
+      config = config.delete_if { |k, _| k =~ /BRIDGE.*/ } if type != "br"
+      config = config.delete_if { |k, _| k =~ /TUNNEL.*/ } if !["tun", "tap"].include?(type)
+      config = config.delete_if { |k, _| k == "VLAN_ID" || k == "ETHERDEVICE" } if type != "vlan"
+      config = config.delete_if { |k, _| k == "IPOIB_MODE" } if type != "ib"
+      config = config.delete_if { |k, _| k == "INTERFACE" } if type != "dummy"
+      config = config.delete_if { |k, _| k == "IFPLUGD_PRIORITY" } if config["STARTMODE"] != "ifplugd"
+
+      # all keys / values has to be strings, agent won't write it otherwise
+      config.map { |k, v| [k, v.to_s] }.to_h
+    end
+
+    # Updates itself according to the given sysconfig configuration
+    #
+    # @param devmap [Hash<String, String>] a key, value map where key is sysconfig option
+    #                                      and corresponding value is the option value
+    def load_sysconfig(devmap)
+      @config = !devmap.nil? ? @config.merge(devmap) : @config
```

Comment:
This is an implementation detail but SCNR
```suggestion
      @config.merge!(devmap || {})
```

Also, make up your mind whether devmap is allowed to be `nil` and update the `@param` or the code.